File: Daily/Rust/Leetcode1442-形成两个异或相等数组的三元组.rs
impl Solution {
    pub fn count_triplets(arr: Vec<i32>) -> i32 {
        let n = arr.len();
        let mut presum = vec![0;n + 1];
        for i in 0..n{
            presum[i + 1] = presum[i] ^ arr[i]
        }
        let mut res = 0;
        for i in 0..n{
            for k in (i + 1)..n{
                if presum[i] == presum[k + 1]{
                    res += k - i;
                }
            }
        }
        res as i32
    }
}

File: Leetcode/sbgo/Leetcode1442-形成两个异或相等数组的三元组.go
package sbgo

func countTriplets(arr []int) int {
	n := len(arr)
	presum := make([]int, n+1)
	for i := 0; i < n; i++ {
		presum[i+1] = presum[i] ^ arr[i]
	}
	res := 0
	for i := 0; i < n; i++ {
		for k := i + 1; k < n; k++ {
			if presum[i] == presum[k+1] {
				res += k - i
			}
		}
	}
	return res
}
